### PR TITLE
Misc code cleanup and handling of analyzer warnings.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/bin/service/search.dart [hunk 1]
```diff
@@ -31,6 +31,7 @@ Future main() async {
   });
   for (int i = 0; i < envConfig.isolateCount; i++) {
     await Isolate.spawn(
+      // ignore: uses_dynamic_as_bottom
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/bin/service/search.dart]: Wrapping here yields an `Isolate.spawn expects to be passed a static or top-level function` exception. wrapping the method...

**Comment by isoos** [comment]
PTAL

File: app/bin/service/search.dart [hunk 2]
```diff
@@ -32,7 +32,7 @@ Future main() async {
   });
   for (int i = 0; i < envConfig.isolateCount; i++) {
     await Isolate.spawn(
-      _main,
```
Comment (kevmoo) [inline app/bin/service/search.dart]: I mean, like an inline lambda `(msg) => _main(msg)`

does that not work?

Comment (isoos) [reply app/bin/service/search.dart]: That gives you an `Isolate.spawn expects to be passed a static or top-level function` exception.

Comment (kevmoo) [reply app/bin/service/search.dart]: Ugh. K

File: app/bin/service/analyzer.dart [hunk 3]
```diff
@@ -30,14 +30,16 @@ Future main() async {
...
-      startIsolates(logger, _runScheduler);
+      startIsolates(logger, _runSchedulerWrapper);
     });
     _registerServices();
     await runAppEngine((HttpRequest request) =>
         shelf_io.handleRequest(request, analyzerServiceHandler));
   });
 }
 
+void _runSchedulerWrapper(message) => _runScheduler(message);
```
Comment (kevmoo) [inline app/bin/service/analyzer.dart]: Put a comment about this referencing https://github.com/dart-lang/sdk/issues/30755

File: app/bin/service/dartdoc.dart [hunk 4]
```diff
@@ -22,13 +22,15 @@ Future main() async {
...
   withAppEngineServices(() async {
-    await startIsolates(logger, _runScheduler);
+    await startIsolates(logger, _runSchedulerWrapper);
     _registerServices();
     await runAppEngine((HttpRequest request) =>
         shelf_io.handleRequest(request, dartdocServiceHandler));
   });
 }
 
+void _runSchedulerWrapper(message) => _runScheduler(message);
```
Comment (kevmoo) [inline app/bin/service/dartdoc.dart]: Put a comment about this referencing dart-lang/sdk#30755

File: app/bin/service/search.dart [hunk 5]
```diff
@@ -41,6 +41,8 @@ Future main() async {
   }
 }
 
+void _mainWrapper(message) => _main(message);
```
Comment (kevmoo) [inline app/bin/service/search.dart]: Put a comment about this referencing dart-lang/sdk#30755

**Pull request merged**